### ISSUE TITLE
snapd-qt fails to link correctly (snapd-glib 1.21)

### ISSUE THREAD

**Issue by Conan-Kudo** [snapcore/snapd-glib#19]
When trying to build snapd-glib 1.21 for Fedora, I get the following error:

```
Making all in snapd-qt
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-alias.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-assertion.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-auth-data.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-app.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-change.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-connection.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-icon.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-plug.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-price.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-request.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-screenshot.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-slot.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-snap.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-client.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      alias_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      app_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      assertion_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      auth-data_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      change_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      client_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      connection_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      icon_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      plug_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      price_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      request_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      screenshot_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      slot_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      snap_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      system-information_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      task_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      client-private_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      wrapped-object_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-system-information.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  GEN      qml-plugin_moc.cpp
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-task.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-stream-wrapper.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-alias_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-app_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-assertion_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-auth-data_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-change_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-connection_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-client_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-icon_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libSnapd_la-qml-plugin.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-plug_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-price_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-request_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-slot_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-screenshot_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-snap_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-system-information_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-task_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-client-private_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libsnapd_qt_la-wrapped-object_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXX      libSnapd_la-qml-plugin_moc.lo
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXXLD    libSnapd.la
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
/usr/bin/ld: cannot find -lsnapd-qt
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:609: libSnapd.la] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Entering directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
  CXXLD    libsnapd-qt.la
make[2]: Leaving directory '/builddir/build/BUILD/snapd-glib-1.21/snapd-qt'
make[1]: *** [Makefile:465: all-recursive] Error 1
make: *** [Makefile:397: all] Error 2
```

Full build logs from the Koji scratch build are available [here](https://koji.fedoraproject.org/koji/taskinfo?taskID=21795551).

**Comment by robert-ancell** [comment]
Looks like it's built the QML plugin before snapd-qt, so must be a missing dependency. It builds in the other order here.

**Comment by robert-ancell** [comment]
Fixed in f092fa0. Did I mention how much I love autotools? ;)

**Comment by Conan-Kudo** [comment]
As my friend @ignatenkobrain would say... Switch to Meson! :)

**Comment by robert-ancell** [comment]
I would switch to Meson but I need to support snapd-glib back to Ubuntu 14.04 LTS...

**Comment by robert-ancell** [comment]
Released with change in snapd-glib 1.22. Don't make me keep making releases just to match the Fedora version number :P

**Comment by ignatenkobrain** [comment]
@robert-ancell why do you need to support such ancient distro?

**Comment by Conan-Kudo** [comment]
@robert-ancell snapd-glib isn't even *available* for Ubuntu 14.04, so I'm not sure how it matters in this?

**Comment by robert-ancell** [comment]
It's in progress:
https://bugs.launchpad.net/bugs/1698984

I wasn't aware that meson is available as far back as xenial through backports. I'll have to investigate if I can rely on that for SRUs.
